widgets/src/interfaces/index.ts

```diff
@@ -27,4 +27,51 @@ export interface CartWidgetParams {
   withPopup?: boolean
 }
 
-export interface CartWidgetRequestBody extends ApiRequestBody, CartWidgetParams {}
+export interface SpendLevelWidgetParams {
+  color: (typeof WIDGET_COLORS.spendLevel)[number]
+  currency: string
+  withPopup?: boolean
+}
+
+export interface PerOrderWidgetParams {
+  color: (typeof WIDGET_COLORS.spendLevel)[number]
+  currency: string
+  withPopup?: boolean
+}
+
+export interface ByPercentageWidgetParams {
+  color: (typeof WIDGET_COLORS.spendLevel)[number]
+  withPopup?: boolean
+}
+
+export interface TieredSpendLevelWidgetParams {
+  color: (typeof WIDGET_COLORS.spendLevel)[number]
+  currency: string
+  withPopup?: boolean
+}
+
+export interface PerProductWidgetParams {
+  color: (typeof WIDGET_COLORS.spendLevel)[number]
+  productId: string
+  withPopup?: boolean
+}
+
+export interface TopStatsWidgetParams {
+  color: (typeof WIDGET_COLORS.spendLevel)[number]
+}
```

Comment:
That wouldn't really fix anything because it would just replace `color: etc.` with `extends ColoredWidget`. Don't really like the approach because it obfuscates the type without giving any value in return